### PR TITLE
Add missing parameters and clean up `add_scalar_bar` docstring

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

`fmt` and `vertical` were missing.  

A few defaults are still missing, because in those cases I wasn't sure what would be the accurate description.  For example, I'm not sure what not setting `n_colors` does; it depends on `vtk` behavior.

I opted to specifically call out when the default inherits from `pyvista.global_theme`, except for `position_x` and `position_y`, which are more complex.

I used the `numpydoc` style preference for when only a set of values is accepted for `font_family`.

See https://numpydoc.readthedocs.io/en/latest/format.html#parameters.

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
I also reordered the parameters to match the order in the function definition

File: pyvista/plotting/scalar_bars.py [hunk 1]
```diff
@@ -123,68 +123,82 @@ def add_scalar_bar(self, title='', mapper=None, n_labels=5, italic=False,
 
         title_font_size : float, optional
             Sets the size of the title font.  Defaults to None and is sized
```
Comment (akaszynski) [inline pyvista/plotting/scalar_bars.py:125]:
```suggestion
            Sets the size of the title font.  Defaults to ``None`` and is sized
```

File: pyvista/plotting/scalar_bars.py [hunk 2]
```diff
@@ -123,68 +123,82 @@ def add_scalar_bar(self, title='', mapper=None, n_labels=5, italic=False,
...
-            automatically.
+            according to ``pyvista.global_theme``.
 
         label_font_size : float, optional
             Sets the size of the title font.  Defaults to None and is sized
```
Comment (akaszynski) [inline pyvista/plotting/scalar_bars.py:129]:
```suggestion
            Sets the size of the title font.  Defaults to ``None`` and is sized
```

File: pyvista/plotting/scalar_bars.py [hunk 3]
```diff
@@ -123,68 +123,82 @@ def add_scalar_bar(self, title='', mapper=None, n_labels=5, italic=False,
...
             * ``color='#FFFFFF'``
 
-        font_family : string, optional
-            Font family.  Must be either courier, times, or arial.
+        font_family : {'courier', 'times', 'arial'}
+            Font family.  Default is set by ``pyvista.global_theme``.
 
         shadow : bool, optional
-            Adds a black shadow to the text.  Defaults to False
+            Adds a black shadow to the text.  Defaults to False.
```
Comment (akaszynski) [inline pyvista/plotting/scalar_bars.py:145]:
```suggestion
            Adds a black shadow to the text.  Defaults to ``False``.
```

**Review by akaszynski — APPROVED** [review]
Made a few nitpicks, but other than that LGTM.

**Comment by adeak** [comment]
I don't want to sidetrack the PR so this is just food for thought for another time: is it not surprising UX that `plotter.add_scalar_bar()` doesn't use the plotter's own theme, but rather always falls back to the global theme?

I know that the reason for this is that the method is actually defined on a separate class, and it gets wrapped by `plotter`'s method. Would it be worth the complexity to go up to `self._plotter` to query the theme in `ScalarBars.add_scalar_bar()`?

**Comment by MatthewFlamm** [comment]
> is it not surprising UX that plotter.add_scalar_bar() doesn't use the plotter's own theme, but rather always falls back to the global theme?

I have not used per-plotter themes (nor did I know this was a possibility) but if I did, I would expect it to use the plotter's theme like you suggest.

I don't see an example that shows how to set a per-plotter theme in the examples or in the module level dosctring of `themes`.

**Comment by adeak** [comment]
It's a new addition, you can find a reference in `BasePlotter`'s docstring [here](https://github.com/pyvista/pyvista/blob/master/pyvista/plotting/plotting.py#L156-L157).

**Comment by MatthewFlamm** [comment]
Same issue here in that Merging is blocked.  I was able to restart the CI due to some azure pip caching problem, so that part is working.

**Pull request merged**